### ISSUE TITLE
Worker gets killed by the OS, miner doesn't detect restart

### ISSUE THREAD

**Issue by karalabe** [filecoin-project/lotus#4676]
My worker has been getting killed a few times per day lately due to going OOM (1 bug). With latest master, the miner doesn't detect this and never relinquishes the failed tasks resources, so it will lock up thinking there's no RAM left:

```
# lotus-miner sealing workers
Worker 842c063c-54ec-4ad0-a1e7-011d40536da9, host filecoin
	CPU:  [                                                                ] 0/48 core(s) in use
	RAM:  [||||||||||||||||||||||||||||||||||||||||||||||||||||||||        ] 87% 442 GiB/503.6 GiB
	VMEM: [||||||||||||||||||||||||||||||||||||||||||||||||||||||||        ] 87% 442 GiB/503.6 GiB
	GMEM: [                                                                ] 0% 0 B/10 GiB

# free
              total        used        free      shared  buff/cache   available
Mem:      528113912     5970436   456589584         116    65553892   522143476
```

**Comment by rjan90** [comment]
On newer versions of Lotus, I think this issue has been addressed? At least on `lotus version 1.11.1-rc1+mainnet+git.193cb9134` the miner detects a restart of a lotus-worker. #rengjøring

**Comment by jennijuju** [comment]
Please open a new ticket if you are still having this issue with the latest lotus
